### PR TITLE
Simple CI with 1 node for metal-k8s

### PR DESCRIPTION
**Pull request by Zempashi**
CI on metal-k8s. Basic test on storage.

https://eve.devsca.com/github/scality/metal-k8s/#/builders/1/builds/51

TODO: 
- since last rebase elasticsearch is constently failing
- make flakyness assessment

### CODE REVIEW

File: eve/main.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          KUBECONFIG: inventories/single-node-ci/artifacts/admin.conf
+    - ShellCommand:
+        name: 'Test storage'
+        haltOnFailure: true
+        command: |-
+            /usr/local/bin/kubectl get pv
+            /usr/local/bin/kubectl apply -f tests/storage/test_pv.yml
+            #retry kubectl get pods test-volume -o jsonpath={.status.phase} | grep -v -E '(Failed|Completed)' ]
+            sleep 10
+            /usr/local/bin/kubectl delete pod test-pv
```
Comment (NicolasT) [inline eve/main.yml]: We're not really checking somehow whether the test ran inside the `Pod` succeeded?

File: eve/main.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            /usr/local/bin/kubectl delete pvc testclaim
+        env:
+          KUBECONFIG: inventories/single-node-ci/artifacts/admin.conf
```
Comment (NicolasT) [inline eve/main.yml]: Consider putting `/usr/local/bin` in `PATH` through `env` such that you can simply run `kubectl`, which will reduce changes required when we do the `make shellrc` thing.

File: eve/main.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    - ShellCommand:
+        name: 'Verify that some pv are in released state'
+        command: |-
+          /usr/local/bin/kubectl get pv
+          [ "$(/usr/local/bin/kubectl get pv | grep 'Released')" ]
+        env:
+          KUBECONFIG: inventories/single-node-ci/artifacts/admin.conf
+    - ShellCommand:
+        name: 'Reclaime storage'
```
Comment (NicolasT) [inline eve/main.yml]: `Reclaim`

File: eve/single_node/group_vars/k8s-cluster/single_node.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  vgs:
+    kubevg:
+      drives: ['/dev/loop0']
+
+metal_k8s_storage_class:
+  storage_classes:
+    local-lvm:
+      is_default: true
+  lvm_conf:
+    default_fstype: 'ext4'
```
Comment (NicolasT) [inline eve/single_node/group_vars/k8s-cluster/single_node.yml]: Must all of these be set when we could rely on defaults?

File: eve/single_node/single_node.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+#! /bin/sh
+
+dd if=/dev/zero of=/var/lib/kube_lvm bs=1M count=20480
```
Comment (NicolasT) [inline eve/single_node/single_node.sh]: `truncate -s 2G /var/lib/kube_lvm` is likely faster and introduces less needless IO (could also use `dd` to create a sparse file, but I always forget what the right arguments for that are :smile:)

File: eve/workers/zenko.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: "zenko-test-pod"
+spec:
+  containers:
+  - name: aggressor
```
Comment (NicolasT) [inline eve/workers/zenko.yaml]: Que?

File: eve/workers/zenko.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    image: zenko/zenko-releng:0.0.7
+    resources:
+      requests:
+        cpu: 500m
+        memory: 1Gi
+      limits:
+        cpu: "1"
+        memory: 2Gi
+    command: ["/bin/sh", "-c", "buildbot-worker create-worker . ${BUILDMASTER}:${BUILDMASTER_PORT} ${WORKERNAME} ${WORKERPASS} && buildbot-worker start --nodaemon"]
```
Comment (NicolasT) [inline eve/workers/zenko.yaml]: A bit odd to sometimes use `/bin/sh` as `command`, then pass `args`, and sometimes do everything in `command`.

File: roles/info_displayer/tasks/main.yml [hunk 8]
```diff
@@ -6,7 +6,7 @@
         {{ hostvars.localhost.displayable_info|default([])
             + add_info }}
     ansible_connection: '{{ hostvars.localhost.ansible_connection }}'
-    ansible_python_interpreter: '{{ hostvars.localhost.ansible_python_interpreter }}'
+    ansible_python_interpreter: '{{ hostvars.localhost.ansible_python_interpreter|default(omit) }}'
```
Comment (NicolasT) [inline roles/info_displayer/tasks/main.yml]: Let's make sure to land the fix in its own PR before merging this, to retain references etc.

File: eve/main.yml [hunk 9]
```diff
@@ -0,0 +1,91 @@
+version: "0.2"
+
+branches:
+  default:
+    stage: "pre-merge"
+
+stages:
+  pre-merge:
```
Comment (tcarmet) [inline eve/main.yml]: @Zempashi  Is there any reason why you're not directly calling the `single-node` stage?

File: eve/main.yml (hunk 1 above)
Comment (Zempashi) [reply eve/main.yml]: Not yet indeed

File: eve/main.yml (hunk 2 above)
Comment (Zempashi) [reply eve/main.yml]: I was thinking about that but I'm not sure I can reference $PATH in an env variable of buildbot (the goal is to add the /usr/local/bin/ not to set PATH=/usr/local/bin)

File: eve/single_node/group_vars/k8s-cluster/single_node.yml (hunk 4 above)
Comment (Zempashi) [reply eve/single_node/group_vars/k8s-cluster/single_node.yml]: but defaults are replaced not merged. We should convert this in a less tree-ish way, more flat. This allow to override just one key.

File: eve/workers/zenko.yaml (hunk 6 above)
Comment (Zempashi) [reply eve/workers/zenko.yaml]: copy-paste, left-over of sylvain PR, but don't what it mean either

File: eve/main.yml (hunk 9 above)
Comment (Zempashi) [reply eve/main.yml]: To let other people adding tests that run in parallel..

Comment (tcarmet) [reply eve/main.yml]: yeah, make sense!

File: eve/main.yml [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+            export PATH=${PATH}:/usr/local/bin
+            make test-simple-storage
+        workdir: build/tests
+        env:
+          KUBECONFIG: inventories/single-node-ci/artifacts/admin.conf
+    - ShellCommand:
+        name: 'Verify that some pv are in released state'
+        command: |-
+          export PATH=${PATH}:/usr/local/bin
+          kubectl get pv
```
[the remark this answers is not in the document]
Comment (Zempashi) [reply eve/main.yml]: After discussion avoid awk to parse the result (because grep 'Released' is too wide), result bakcin two commands
